CPP/DesktopToastsCppApp/DesktopNotificationManagerCompat.h

```diff
@@ -0,0 +1,108 @@
+    /// under Desktop Bridge. Call this upon application startup, before calling any other APIs.
+    /// </summary>
+    /// <param name="aumid">An AUMID that uniquely identifies your application.</param>
+    /// <param name="clsid">The CLSID of your NotificationActivator class.</param>
+    void RegisterAumidAndComServer(const wchar_t *aumid, GUID clsid);
+
+    /// <summary>
+    /// Registers your module to handle COM activations. Call this upon application startup.
+    /// </summary>
+    void RegisterActivator();
+
+    /// <summary>
+    /// Creates a toast notifier. You must have called RegisterActivator first (and also RegisterAumidAndComServer if you're a classic Win32 app), or this will throw an exception.
+    /// </summary>
+    Microsoft::WRL::ComPtr<IToastNotifier> CreateToastNotifier();
+
+    /// <summary>
+    /// Creates a toast notification. This is simply a convenience helper method.
+    /// </summary>
+    Microsoft::WRL::ComPtr<IToastNotification> CreateToastNotification(ABI::Windows::Data::Xml::Dom::IXmlDocument* content);
+
+    /// <summary>
+    /// Creates an XmlDocument initialized with the specified string. This is simply a convenience helper method.
+    /// </summary>
+    Microsoft::WRL::ComPtr<ABI::Windows::Data::Xml::Dom::IXmlDocument> CreateXmlDocumentFromString(const wchar_t *xmlString);
+
+    /// <summary>
+    /// Gets the DesktopNotificationHistoryCompat object. You must have called RegisterActivator first (and also RegisterAumidAndComServer if you're a classic Win32 app), or this will throw an exception.
+    /// </summary>
+    std::unique_ptr<DesktopNotificationHistoryCompat> get_History();
+
+    /// <summary>
+    /// Gets a boolean representing whether http images can be used within toasts. This is true if running under Desktop Bridge.
+    /// </summary>
+    bool CanUseHttpImages();
+}
+
+class DesktopNotificationHistoryCompat
+{
+public:
+
+    /// <summary>
+    /// Removes all notifications sent by this app from action center.
+    /// </summary>
+    void Clear();
+
+    /// <summary>
+    /// Gets all notifications sent by this app that are currently still in Action Center.
+    /// </summary>
+    Microsoft::WRL::ComPtr<ABI::Windows::Foundation::Collections::IVectorView<ToastNotification*>> GetHistory();
+
+    /// <summary>
+    /// Removes an individual toast, with the specified tag label, from action center.
+    /// </summary>
+    /// <param name="tag">The tag label of the toast notification to be removed.</param>
+    void Remove(const wchar_t *tag);
+
+    /// <summary>
+    /// Removes a toast notification from the action using the notification's tag and group labels.
+    /// </summary>
+    /// <param name="tag">The tag label of the toast notification to be removed.</param>
+    /// <param name="group">The group label of the toast notification to be removed.</param>
+    void Remove(const wchar_t *tag, const wchar_t *group);
```

Comment:
How does this remove? only if matches tag and group? either? Not a fan of just calling this "Remove", it's not really clear what arg is being passed in. Maybe something like RemoveByTag, RemoveByGroup, RemoveByTagAndGroup

Reply:
Ah I suppose I should follow the platform C++ naming, which is `RemoveGroupedTag`... not a great name, but consistency with the platform APIs is helpful.

I was following the C# naming conventions of the platform, which just leaves it as an overload without a different name.